internal/pkg/storageos/volume_labels.go

```diff
@@ -162,3 +180,77 @@ func (c *Client) EnsureReplicas(ctx context.Context, key client.ObjectKey, desir
 	}
 	return observeErr(nil)
 }
+
+// EnsureFailureMode ensures that the desired failure mode has been applied to
+// the StorageOS volume.
+func (c *Client) EnsureFailureMode(ctx context.Context, key client.ObjectKey, desired string) error {
+	funcName := "ensure_failure_mode"
+	start := time.Now()
+	defer func() {
+		metrics.Latency.Observe(funcName, time.Since(start))
+	}()
+	observeErr := func(e error) error {
+		metrics.Errors.Increment(funcName, e)
+		return e
+	}
+
+	ctx = c.AddToken(ctx)
+
+	vol, err := c.getVolume(ctx, key)
+	if err != nil {
+		return observeErr(err)
+	}
+
+	var current string
+
+	// If label already set, get value.
+	for k, v := range vol.Labels {
```

Comment:
I guess `current := vol.Labels[ReservedLabelFailureMode]` does the same.
`var current string` is an empty string while `vol.Labels[ReservedLabelFailureMode]` returns value or empty string.

Reply:
yes...  I do this often - I don't know why!